### PR TITLE
Fixes for stuck builds

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

Fixes for stuck builds 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: tests/server/test_compilerservice.py [hunk 1]
```diff
@@ -975,6 +973,8 @@ async def run_compile_and_wait_until_compile_is_done(
     run._make_pull_fail = fail_on_pull
     run.block = False
 
+    current_task = compiler_service._env_to_compile_task[env_id]
```
Comment (wouterdb) [inline tests/server/test_compilerservice.py:976]: this could produce an index exception in rare cases

Comment (arnaudsjs) [inline tests/server/test_compilerservice.py:976]: This needs to be above the `run.block = False` statement.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: changelogs/unreleased/test_fixes.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+description: Small test fixes
+change-type: patch
+destination-branches: [master]
```
[the remark this answers is not in the document]
Comment (wouterdb) [reply changelogs/unreleased/test_fixes.yml:3]:
```suggestion
destination-branches: [master, iso7]
```

File: tests/server/test_compilerservice.py (hunk 1 above)
Comment (wouterdb) [inline tests/server/test_compilerservice.py:976]:
```suggestion
    run._make_pull_fail = fail_on_pull
    
    current_task = compiler_service._env_to_compile_task[env_id]
    run.block = False
```

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**